docs/docs/detectors/13-unused-return-enum.md

```diff
@@ -1,41 +1,42 @@
 # Unused return enum
 
-### What it does
+## Description
 
-It warns if a function that returns a Result type does not return the Result enum variant (Ok/Err).
+- Vulnerability Category: `Validations and error handling`
+- Vulnerability Severity: `Minor`
+- Detectors: [`unused-return-enum`](https://github.com/CoinFabrik/scout-soroban/tree/main/detectors/unused-return-enum)
+- Test Cases: [`unused-return-enum-1`](https://github.com/CoinFabrik/scout-soroban/tree/main/test-cases/unused-return-enum/unused-return-enum-1) [`unused-return-enum-2`](https://github.com/CoinFabrik/scout-soroban/tree/main/test-cases/unused-return-enum/unused-return-enum-2)
+ 
+Soroban	 messages can return a `Result` enum with a custom error type. This is useful for the caller to know what went wrong when the message fails.
 
-### Why is this bad?
+The definition in Rust of the `Result` enum is:
 
-If any of the variants (Ok/Err) is not used, the code could be simplified or it could imply a bug.
+```rust
+enum Result<T, E> {
+    Ok(T),
+    Err(E),
+}
+```
 
-### Known problems
+## Why is this bad?
 
-If definitions of `Err()` and/or `Ok()` are in the code but do not flow to the return value due to the definition of a variable or because they are defined in a dead code block, the warning will not be shown. If the definitions are made in an auxiliary method, the warning will be shown, resulting in a false positive.
+If any of the variants (Ok/Err) is not used, the code could be simplified or it could imply a bug.
```

Comment:
If either variant (`Ok` or `Err`) is not used in the code, it could indicate that the `Result` type is unnecessary and that the code could be simplified. Alternatively, it might suggest a bug where a possible outcome is not being handled properly.